internal/lua/stringlib.go

```diff
@@ -311,12 +311,45 @@ func stringFormat(ctx context.Context, l *State) (int, error) {
 			if err != nil {
 				return 0, err
 			}
-			if c == 'a' || c == 'A' {
-				// Hexadecimal float. Go uses 'x'/'X'.
-				spec = spec[:len(spec)-1] + string(c+('X'-'A'))
+
+			isUpper := (c >= 'A' && c <= 'Z')
+			if math.IsNaN(n) {
```

Comment:
nit: mind changing this `if`/`else-if` into a `switch`?

Reply:
How would I turn this if else into a switch? From what I can see, switch only check one value at a time.

This is my first time writing Go code, so any advice is welcome.